### PR TITLE
Fix ThroughputResponse.IsReplacePending

### LINKED ISSUE #1024: ThroughputResponse reads wrong header for IsReplacePending
**Describe the bug**
`ThroughputResponse.IsReplacePending` currently evaluates to a `FormatException` as it reads from the `WFConstants.BackendHeaders.MinimumRUsForOffer` header

**To Reproduce**
Scale the database to a value that requires more allocation at the backend, attempt to read `ThroughputResponse.IsReplacePending` flag

**Expected behavior**
`True` or `False` value to be returned

**Actual behavior**
`System.FormatException` is thrown

**Environment summary**
SDK Version: 3.4.1
OS Version (e.g. Windows, Linux, MacOSX) Windows

### PR DESCRIPTION
**Pull request by wahyuen**
fix the value that is being retrieved when mapping ThoughputResponse.IsReplacePending

# Pull Request Template

## Description

This PR is to fix the value that is being retrieved when mapping ThoughputResponse.IsReplacePending as it is currently incorrectly attempting to read `WFConstants.BackendHeaders.MinimumRUsForOffer`

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Closing issues
Closes #1024

### CODE REVIEW

File: Microsoft.Azure.Cosmos/src/Resource/Throughput/ThroughputResponse.cs [hunk 1]
```diff
@@ -82,7 +82,7 @@ protected ThroughputResponse()
             {
                 if (this.Headers.GetHeaderValue<string>(WFConstants.BackendHeaders.OfferReplacePending) != null)
                 {
-                    return Boolean.Parse(this.Headers.GetHeaderValue<string>(WFConstants.BackendHeaders.MinimumRUsForOffer));
+                    return Boolean.Parse(this.Headers.GetHeaderValue<string>(WFConstants.BackendHeaders.OfferReplacePending));
```
Comment (j82w) [inline Microsoft.Azure.Cosmos/src/Resource/Throughput/ThroughputResponse.cs]: Please add tests and update the changelog

Comment (wahyuen) [reply Microsoft.Azure.Cosmos/src/Resource/Throughput/ThroughputResponse.cs]: added and updated :)

**Pull request merged**